packages/ab-testing/src/index.ts

```diff
@@ -25,29 +30,76 @@ function getModuloValue(experiment: string, userId: number | string): number {
     return crc32.calculate(String(userId), crc32.calculate(experiment)) % 100;
 }
 
+function validateAllocationCriteria(criteria: ForceInclude, userProfile: UserProfile): boolean {
+    let itemMatchesCriteria = true;
+    for (const key in criteria) {
+        if (!criteria[key].includes(userProfile[key])) {
+            itemMatchesCriteria = false;
+        }
+    }
+
+    return itemMatchesCriteria;
+}
+
+function validateUserWithinAllocationRange(userSegmentNum: number, range: AllocationV1) {
```

Comment:
TBH I'm not a fan of these abstractions.

The LoC is more than doubled than before without gaining any benefits.

Reply:
I gotta agree with Tao here, the names are quite similar and I think inlining them make the readability of the code better